### PR TITLE
F2F-375: Added dynamodb template as well as workflow files

### PR DESCRIPTION
**Pull request by AyeshaKhanGDS**
Added infra-l2-dynamo folder with package.json files and template file, as well as workflow file

### CODE REVIEW

File: infra-l2-dynamo/template.yaml [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    DeletionPolicy: Delete
+    UpdateReplacePolicy: Delete
+    Properties:
+      TableName: !Sub "session-events-${AWS::StackName}"
+      BillingMode: "PAY_PER_REQUEST"
+      AttributeDefinitions:
+        - AttributeName: "userId"
+          AttributeType: "S"
+        - AttributeName: "authorizationCode"
+          AttributeType: "S"
```
Comment (darrendunford) [inline infra-l2-dynamo/template.yaml:72]:
```suggestion
```
Not required as no index is required on `authorizationCode`

File: infra-l2-dynamo/template.yaml [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+          KeySchema:
+            - AttributeName: "authorizationCode"
+              KeyType: "HASH"
+          Projection:
+            NonKeyAttributes:
+              - "userId"
+              - "redirectUri"
+              - "clientId"
+              - "authSessionState"
+            ProjectionType: "INCLUDE"
```
Comment (darrendunford) [inline infra-l2-dynamo/template.yaml:87]:
```suggestion
```
Not required

File: infra-l2-dynamo/template.yaml [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+        PointInTimeRecoveryEnabled: true
+      SSESpecification:
+        KMSMasterKeyId: !Ref SessionEventsTableKey
+        SSEEnabled: true
+        SSEType: KMS
+
+  SessionEventsTableKey:
+    Type: AWS::KMS::Key
+    Properties:
+      Description: !Sub "${AWS::StackName} session table key"
```
Comment (darrendunford) [inline infra-l2-dynamo/template.yaml:101]:
```suggestion
      Description: !Sub "${AWS::StackName} sessionEvents table key"
```
nit

File: infra-l2-dynamo/src/tests/infra/infra.test.ts [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+// https://docs.aws.amazon.com/cdk/v2/guide/testing.html <--- how to use this file
+
+let template: Template;
+
+beforeAll(() => {
+  let yamlTemplate: any = load(readFileSync('./template.yaml', 'utf-8'), { schema: schema })
+  template = Template.fromJSON(yamlTemplate)
+})
+
+it("should define 2 DynamoDB tables", async () => {
```
Comment (darrendunford) [inline infra-l2-dynamo/src/tests/infra/infra.test.ts:15]:
```suggestion
it("should define 1 DynamoDB table", async () => {
```

**Review by darrendunford — CHANGES_REQUESTED** [review]

File: infra-l2-dynamo/src/tests/infra/infra.test.ts (hunk 4 above)
Comment (AyeshaKhanGDS) [reply infra-l2-dynamo/src/tests/infra/infra.test.ts]: Adjusted

File: infra-l2-dynamo/template.yaml (hunk 1 above)
Comment (AyeshaKhanGDS) [reply infra-l2-dynamo/template.yaml:72]: removed

File: infra-l2-dynamo/template.yaml (hunk 2 above)
Comment (AyeshaKhanGDS) [reply infra-l2-dynamo/template.yaml:87]: removed

File: infra-l2-dynamo/template.yaml (hunk 3 above)
Comment (AyeshaKhanGDS) [reply infra-l2-dynamo/template.yaml:101]: adjusted

**Pull request merged**